### ISSUE TITLE
Add support for array column type in PostgreSQL client (Scala collection types)

### ISSUE THREAD

**Issue by pmlopes** [vert-x3/vertx-mysql-postgresql-client#73]
@jvz commented on [Thu Mar 30 2017](https://github.com/vert-x3/issues/issues/251)

If I have a table like this:

```
CREATE TABLE example (
  id UUID PRIMARY KEY,
  tags TEXT[]
);
```

Then when I query for rows, `Json.checkAndCopy()` throws an IllegalStateException because it doesn't know how to handle the class `scala.collection.mutable.ArrayBuffer`. Now I'm not super experienced with Scala at all, but it seems like this can be supported by adding generic support for `scala.collection.Seq[T]`, however, it might be nicer to support `Traversable[T]` since that seems to be one of the most generic collection classes in Scala.

---

@EmadAlblueshi commented on [Sun Apr 02 2017](https://github.com/vert-x3/issues/issues/251#issuecomment-290981645)

Kindly take a look https://github.com/vert-x3/vertx-sql-common/issues/17

---

@jvz commented on [Sun Apr 02 2017](https://github.com/vert-x3/issues/issues/251#issuecomment-291000396)

I don't recall having a problem inserting the data (I inserted as a `List` instance using `queryWithParams`), but I had issues retrieving it since the JsonObject/JsonArray classes were unaware of how to convert a Scala collection into an appropriate type.

**Comment by jan-zajic** [comment]
Current issue when query array:
java.lang.IllegalStateException: Illegal type in JsonObject: class scala.collection.mutable.ArrayBuffer
	at io.vertx.core.json.Json.checkAndCopy(Json.java:125)
	at io.vertx.core.json.JsonArray.add(JsonArray.java:435)
	at io.vertx.ext.asyncsql.impl.ScalaUtils.convertValue(ScalaUtils.java:128)
	at io.vertx.ext.asyncsql.impl.ScalaUtils.access$0(ScalaUtils.java:114)
	at io.vertx.ext.asyncsql.impl.ScalaUtils$4.apply(ScalaUtils.java:107)
	at io.vertx.ext.asyncsql.impl.ScalaUtils$4.apply(ScalaUtils.java:1)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1406)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at com.github.mauricio.async.db.general.ArrayRowData.foreach(ArrayRowData.scala:21)
	at io.vertx.ext.asyncsql.impl.ScalaUtils.rowToJsonArray(ScalaUtils.java:104)
	at io.vertx.ext.asyncsql.impl.AsyncSQLConnectionImpl$1.apply(AsyncSQLConnectionImpl.java:336)
	at io.vertx.ext.asyncsql.impl.AsyncSQLConnectionImpl$1.apply(AsyncSQLConnectionImpl.java:1)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1406)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at com.github.mauricio.async.db.general.MutableResultSet.foreach(MutableResultSet.scala:27)
	at io.vertx.ext.asyncsql.impl.AsyncSQLConnectionImpl.rowDataSeqToJsonArray(AsyncSQLConnectionImpl.java:333)
	at io.vertx.ext.asyncsql.impl.AsyncSQLConnectionImpl.queryResultToResultSet(AsyncSQLConnectionImpl.java:307)
	at io.vertx.ext.asyncsql.impl.AsyncSQLConnectionImpl.lambda$10(AsyncSQLConnectionImpl.java:277)
	at io.vertx.core.impl.FutureImpl.tryComplete(FutureImpl.java:126)
	at io.vertx.core.impl.FutureImpl.complete(FutureImpl.java:88)
	at io.vertx.ext.asyncsql.impl.ScalaUtils$1.apply(ScalaUtils.java:53)
	at io.vertx.ext.asyncsql.impl.ScalaUtils$1.apply(ScalaUtils.java:1)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at io.vertx.ext.asyncsql.impl.VertxEventLoopExecutionContext.lambda$0(VertxEventLoopExecutionContext.java:59)
	at io.vertx.core.impl.ContextImpl.lambda$1(ContextImpl.java:337)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:445)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:745)

**Comment by jan-zajic** [comment]
@pmlopes i Fixed it in PR #75
